blog/2018/12/15/practical-web-content-munging/index.markdown

```diff
@@ -0,0 +1,185 @@
+---
+status: published
+title: Practical Web Content Munging
+author: Joe Cooper
+images:
+  banner:
+    src: '/blog/2018/12/14/practical-web-content-munging/banner.jpg'
+    alt: 'An eyeball of alarming size'
+    data:
+      attribution: |-
+        Banner image: [An eyeball looking at Dallas](https://flic.kr/p/Rqj9Jj) by [Joe Cooper](https://www.flickr.com/photos/nerdnomad/) licensed [CC BY-NC-SA 2.0](https://creativecommons.org/licenses/by-nc-sa/2.0/)
+tags:
+  - advent
+  - css
+  - html
+data:
+  bio: swelljoe
+  description: Working with ugly old websites using Mojo::UserAgent and Mojo::DOM
+---
+Following brian d foy's great write-up of using Mojo::DOM selectors from Day 5, I thought it'd be fun to talk about some website migration scripts I recently built using Mojo::UserAgent and Mojo::DOM, two powerful aspects of Mojolicious. I've never used Mojo for a project before, but when I started looking at updating an old website that hasn't been updated in roughly 15 years (it's gotten new content, but the design is unchanged) Mojo::UserAgent and Mojo::DOM looked great. In the past I would have used regexes, and then I would likely have had to add a bunch of special cases where the regex didn't quite match, and then after a few hours of fighting I would have resigned myself to cleaning up the remaining handful of broken pages by hand. No more! Mojo::DOM always gets it right, and with alarmingly little code.
+---
+
+##From Static Site to Static Site Generator
+
+The problem I set out to solve was taking an old static website that was once hosted on SourceForge.net and migrating it to an exciting new...um...static website. But, this time, it'll be a modern take on a static website. Instead of editing HTML by hand and using home-built page munging scripts that would do things like insert news items or changelog entries at the top of the content div using regexes, I'll be using a modern static website generator. There are several such generators, including the well-known Jekyll, which is written in Ruby, Hugo, built with Go, and Statocles, which is in Perl and runs this site. For my project, I chose Hugo, for its speed and maturity.
+
```

Comment:
add a "fold" here by adding a line containing only `---`. On the list pages the article will stop here and say "read more" with a link.

Reply:
Also link to Hugo and perhaps a Hugo post example. Maybe add a sentence like "Here is an example Hugo-formatted post, which will be my desired output". Alternatively you might add that to the next paragraph which lays out what you're trying to do.